File: fem/C++/openacc/auto.opt/src/output_ucd.c
/**
 ** OUTPUT_UCD
 **/
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>
#include "pfem_util.h"

extern FILE *fp_log;
//extern void JACOBI(); 
void OUTPUT_UCD()
{
  FILE *fp;
  int i,ie;
  int N0,N1;
  int in1,in2,in3,in4,in5,in6,in7,in8;
  KREAL XX,YY,ZZ;
  /*
    int N3,N4;
    KREAL ZERO;
    KREAL igmax;
  */
  
  char ETYPE[7];
 
/***
	+----------+
	| AVS file |
	+----------+
***/
  if( (fp=fopen("test.inp","w")) == NULL){
    fprintf(stdout,"output file cannot be opened!\n");
    exit(1);}
  
  N0= 0;
  N1= 1;
  //  ZERO= 0.e0;
  
  fprintf(fp,"%8d%8d%8d%8d%8d\n",NP,ICELTOT,N1,N0,N0);
  
  for(i=0;i<NP;i++){
    XX=XYZ[i*3+0];
    YY=XYZ[i*3+1];
    ZZ=XYZ[i*3+2];
    fprintf(fp,"%8d%16.6e%16.6e%16.6e\n",i+1,XX,YY,ZZ);
  }
  
  for(ie=0;ie<ICELTOT;ie++){
    strcpy(ETYPE," hex  ");
    in1= ICELNOD[ie*8+0];
    in2= ICELNOD[ie*8+1];
    in3= ICELNOD[ie*8+2];
    in4= ICELNOD[ie*8+3];
    in5= ICELNOD[ie*8+4];
    in6= ICELNOD[ie*8+5];
    in7= ICELNOD[ie*8+6];
    in8= ICELNOD[ie*8+7];
    fprintf(fp,"%8d%3d%-8s%8d%8d%8d%8d%8d%8d%8d%8d\n",
	    ie+1,N1,ETYPE,in1, in2, in3, in4, in5, in6, in7, in8);
  }
  
  fprintf(fp,"%3d%3d\n",N1, N1);
  fprintf(fp,"temp,temp\n");
  
  for(i=0;i<NP;i++){
    fprintf(fp,"%8d%16.6e\n",i+1,X[i]);
  }
  fclose(fp);
}

File: fem/F/do-concurrent/auto.def/src/output_ucd.f
      subroutine OUTPUT_UCD 
      use pfem_util
      implicit none

      integer(kind=kint) :: i,ie
      integer(kind=kint) :: N0,N1
      integer(kind=kint) :: in1,in2,in3,in4,in5,in6,in7,in8
      real(kind=kreal)   :: XX,YY,ZZ
      character(len=6) :: ETYPE

!C
!C +----------+
!C | AVS file |
!C +----------+
!C===
        open (21 ,file= 'test.inp', status='unknown')

        N0= 0
        N1= 1

        write (21,'(10i8)')  NP, ICELTOT, N1, N0, N0
        do i= 1, NP
          XX= XYZ(i,1)
          YY= XYZ(i,2)
          ZZ= XYZ(i,3)
          write (21,'(i8,3(1pe16.6))') i, XX, YY, ZZ
        enddo
        do ie= 1, ICELTOT
          ETYPE= 'hex   '
          in1= ICELNOD(ie,1)
          in2= ICELNOD(ie,2)
          in3= ICELNOD(ie,3)
          in4= ICELNOD(ie,4)
          in5= ICELNOD(ie,5)
          in6= ICELNOD(ie,6)
          in7= ICELNOD(ie,7)
          in8= ICELNOD(ie,8)

          write (21,'(i8,i3,1x,a6,1x,8i8)')                             &
     &      ie, N1, ETYPE, in1, in2, in3, in4, in5, in6, in7, in8

        enddo

        write (21,'(10i3)')  N1, N1
        write (21,'(a  )') 'temperature,temperature'

        do i= 1, NP
          write (21,'(i8, 4(1pe16.6))')  i, X(i)
        enddo
        close (21)
!C===
      end subroutine output_ucd

